scipy/_lib/doccer.py

```diff
@@ -54,24 +67,25 @@ def docformat(docstring, docdict=None):
         icount = 0
     else:
         icount = indentcount_lines(lines[1:])
-    indent = ' ' * icount
+    indent = " " * icount
     # Insert this indent to dictionary docstrings
     indented = {}
     for name, dstr in docdict.items():
         lines = dstr.expandtabs().splitlines()
         try:
             newlines = [lines[0]]
             for line in lines[1:]:
-                newlines.append(indent+line)
-            indented[name] = '\n'.join(newlines)
+                newlines.append(indent + line)
+            indented[name] = "\n".join(newlines)
         except IndexError:
             indented[name] = dstr
     return docstring % indented
 
 
-def inherit_docstring_from(cls):
-    """
-    This decorator modifies the decorated function's docstring by
+def inherit_docstring_from(
+    cls: type[Any] | Any,
```

Comment:
The [typing docs](https://typing.readthedocs.io/en/latest/spec/concepts.html#union-types) specify `T | Any` to mean something very specific:

> The union `T | Any` is not reducible to a simpler form. It represents an unknown static type with lower bound `T`. That is, it represents an unknown set of objects which may be as large as `object`, or as small as `T`, but no smaller.

But I don't think that mypy or pyright currently implement it this way, and consider `T | Any` equivalent to `Any`. 
However, this might change in the near future. So that's why I tend to avoid using `T | Any`. 

---

As a sidenote; there are more ways in which `Any` can lead to unexpected outcomes, so in my own projects I usually prefer `object` if it's possible. In this case, it shouldn't matter too much; this mostly concerns return types and type arguments.